### PR TITLE
Deserialize stream

### PR DESCRIPTION
**Pull request by chris-smith**
Built on my other pull request - testing and bug fixes for DeserializeStream.

### CODE REVIEW

File: test/DeserializeStream.js [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    deserializeStream.on('message', (message) => {
+      ++iter;
+      expect(message.length).to.equal(len);
+
+      if (iter === numBufs) {
+        done();
+      }
+    });
+
+    // return a random integer between 16 and 28
```
Comment (IanTheEngineer) [inline test/DeserializeStream.js]: Wouldn't this return a random integer between **20** and 28?

File: test/DeserializeStream.js [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+    const buf = new Buffer(len + 5).fill(0);
+    buf[0] = 1;
+    buf.writeUInt32LE(19, 1);
+
+    const bufArr = new Array(numBufs).fill(buf);
+    let bigBuff = Buffer.concat(bufArr);
+
+    deserializeStream.setServiceRespDeserialize();
+    // return a random integer between 16 and 28
```
Comment (IanTheEngineer) [inline test/DeserializeStream.js]: Same question as above.

**Comment by IanTheEngineer** [comment]
Looks good. Consider rebasing off master once #1 is merged, but I think this is good to go :shipit:

File: test/DeserializeStream.js (hunk 1 above)
Comment (chris-smith) [inline test/DeserializeStream.js]: gah yes

File: test/DeserializeStream.js (hunk 2 above)
Comment (chris-smith) [inline test/DeserializeStream.js]: blergh

**Pull request merged**